### PR TITLE
Query pipes

### PR DESCRIPTION
**Pull request by ggurkal**
* Introduce new exception `BadRequestException`.
* Remove default parsing for `number` and `boolean` types.
* Introduce the ability to add pipes to the `Query` decorator.
* Throw in pipes when the value is invalid.
* Fix tests.

### CODE REVIEW

File: lib/pipes/parseBoolean.pipe.spec.ts [hunk 1]
```diff
@@ -5,5 +5,5 @@ describe('ParseBooleanPipe', () => {
 
   it('Should parse the given string as boolean (false)', () => expect(ParseBooleanPipe('false')).toStrictEqual(false));
 
-  it('Should return the given string as string', () => expect(ParseBooleanPipe('test')).toStrictEqual('test'));
+  it('Should return the given string as string', () => expect(() => ParseBooleanPipe('test')).toThrow());
```
Comment (leeuwis) [inline lib/pipes/parseBoolean.pipe.spec.ts:8]: Test description should be updated according to change

File: lib/pipes/parseBoolean.pipe.ts [hunk 2]
```diff
@@ -1,5 +1,13 @@
+import { BadRequestException } from '../exceptions';
+
 export function ParseBooleanPipe(value: any): boolean {
-  return typeof value === 'string' && (value === 'true' || value === 'false')
-    ? value.toLowerCase().trim() === 'true'
-    : value;
+  if (value === true || value === 'true') {
```
Comment (leeuwis) [inline lib/pipes/parseBoolean.pipe.ts:4]: Consideration, do we also want to allow `1` and `'1'`

File: lib/pipes/parseBoolean.pipe.ts [hunk 3]
```diff
@@ -1,5 +1,13 @@
...
+    return true;
+  }
+
+  if (value === false || value === 'false') {
```
Comment (leeuwis) [inline lib/pipes/parseBoolean.pipe.ts:8]: Consideration, do we also want to allow `0` and `'0'`

File: lib/pipes/parseNumber.pipe.spec.ts [hunk 4]
```diff
@@ -6,5 +6,5 @@ describe('ParseNumberPipe', () => {
   it('Should parse the given string as number with decimal points', () =>
     expect(ParseNumberPipe('07.99')).toStrictEqual(7.99));
 
-  it('Should return the given string as string', () => expect(ParseNumberPipe('test')).toStrictEqual('test'));
+  it('Should return the given string as string', () => expect(() => ParseNumberPipe('test')).toThrow());
```
Comment (leeuwis) [inline lib/pipes/parseNumber.pipe.spec.ts:9]: Test description should be updated according to change

**Comment by leeuwis** [comment]
Closes #2 #

File: lib/pipes/parseBoolean.pipe.ts (hunk 2 above)
Comment (ggurkal) [reply lib/pipes/parseBoolean.pipe.ts:4]: I'd say no.

File: lib/pipes/parseBoolean.pipe.ts (hunk 3 above)
Comment (ggurkal) [reply lib/pipes/parseBoolean.pipe.ts:8]: I'd say no.

**Pull request merged**